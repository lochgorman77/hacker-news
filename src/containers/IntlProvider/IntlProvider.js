import React from 'react';
import PropTypes from 'prop-types';
import { IntlProvider as IP } from 'react-intl';

import translations from './translations/translations';

export class IntlProvider extends React.PureComponent {
  render() {
    const { locale } = this.props;
    return (
      <IP locale={locale.code} key={locale.code} messages={translations[locale.code]}>
        {React.Children.only(this.props.children)}
      </IP>
    );
  }
}
IntlProvider.propTypes = {
  locale: PropTypes.object,
  messages: PropTypes.array,
  children: PropTypes.element.isRequired,
};
export default IntlProvider;
