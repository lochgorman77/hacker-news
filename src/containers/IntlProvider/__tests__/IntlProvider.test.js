import React from 'react';
import { IntlProvider } from '../IntlProvider';

const title = React.createElement('div', { key: 1 }, 'test');

const props = {
  locale: { code: 'en' },
  children: title,
};

describe('<IntlProvider />', () => {
  let shallow, wrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<IntlProvider {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper.name()).toBe('IntlProvider');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
