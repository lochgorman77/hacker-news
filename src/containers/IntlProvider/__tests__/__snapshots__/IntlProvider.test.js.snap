// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntlProvider /> renders correctly 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  key="en"
  locale="en"
  messages={
    Object {
      "detail.view.by": "by",
      "detail.view.comments": "comment",
      "detail.view.link": "View Article",
      "menu.link.home": "Hacker News",
      "menu.link.login": "Login",
      "menu.link.new": "New",
      "menu.link.top": "Top",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <div
    key="1"
  >
    test
  </div>
</IntlProvider>
`;
